crates/viewer/re_global_context/src/lib.rs

```diff
@@ -67,6 +67,11 @@ pub enum DisplayMode {
     /// The settings dialog for application-wide configuration.
     Settings,
 
+    // TODO(isse): Use this to know what to switch to when sources are
+    // done loading.
```

Comment:
not sure I parse the intention correctly, you mean this state should know its destination in the future, i.e. something like this?
```
Loading {
    message: String,
    target: Box<DisplayMode>,
}
```
(would make sense to me :))

Reply:
reading on, I realize that this string isn't actually a message but rather the target URL :O
At the very least it should be documented! Ideally we'd be doing this typed, but I figure we don't have the right type here (and decoupling is worth _something_ ;))

Reply:
Yeah this is even further down than `re_viewer_context`, should definitely add documentation.

> you mean this state should know its destination in the future

Yeah, so we only switch to the thing we've loaded if we're still on that page. And not if they have pressed to go to some other recording etc.

The problem with `Box<DisplayMode>` is I think again that the thing we're loading hasn't gotten a `StoreId` yet. But maybe it would be possible to change the order around to have that ahead of time?